.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Cache build artifacts
+
+on:
+  workflow_call:
+
+env:
+  NEXT_PUBLIC_INFURA_KEY: ${{ secrets.NEXT_PUBLIC_INFURA_KEY }}
+  NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID: ${{ secrets.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID }}
+  THE_GRAPH_NETWORK_API_KEY: ${{ secrets.THE_GRAPH_NETWORK_API_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  check-build-cache:
+    name: Check cache for build artifacts
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Check cache for build artifacts
+      id: cache
+      uses: actions/cache/restore@v4
+      with:
+        path:  |
+          ./packages/arb-token-bridge-ui/build
+        key: build-artifacts-${{ github.run_id }}
```

Comment:
we try to find it from cache first, and if it's not there, then we run build, otherwise just use the cache

this is similar to the approach we use for node modules in our actions repo